src/algebra/monoid_algebra.lean

```diff
@@ -1170,6 +1170,44 @@ See note [partially-applied ext lemmas]. -/
   f = g :=
 ring_hom_ext (ring_hom.congr_fun h₁) (monoid_hom.congr_fun h_of)
 
+section opposite
+
+variables {I : Type*} [semiring R]
```

Comment:
I think it would be best to continue using `k` and `G` as in the rest of the file.